src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -116,11 +116,28 @@ private[xml] object TypeCast {
   )
 
   private def parseXmlTimestamp(value: String, options: XmlOptions): Timestamp = {
-    val formatters = options.timestampFormat.map(DateTimeFormatter.ofPattern).
-      map(supportedXmlTimestampFormatters :+ _).getOrElse(supportedXmlTimestampFormatters)
-    formatters.foreach { format =>
+    supportedXmlTimestampFormatters.foreach { format =>
+      try {
+        return Timestamp.from(Instant.from(format.parse(value)))
+      } catch {
+        case _: Exception => // continue
+      }
+    }
+    options.timestampFormat.foreach { formatString =>
+      // Check if there is offset or timezone and apply Spark timeZone if not
+      val hasTemporalInformation = formatString.indexOf("V") +
```

Comment:
I'm not sure we need this - I found that the docs for "withZone" say that it's ignored if the pattern contains timezone info. So it seems like it will just be a default. OK to write a test for that though!

Reply:
Yes I also saw this, but then I encountered some problems between Java 8 and Java 11. We can see in Java 8 [here](https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/java/time/format/Parsed.java#L586) that zone is prioritized over offset, but in Java 9 [here](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/time/format/Parsed.java#L595), offset is prioritized over zone. I made an example below to see the differences:
- Java 8: ![image](https://user-images.githubusercontent.com/29867169/210255492-c304a6bc-dfdb-4121-91fb-553456d6d5ff.png)
- Java 11: ![image](https://user-images.githubusercontent.com/29867169/210255535-f8459988-639a-41ff-9df4-f68172a380bb.png)

So I don't think we can always apply `withZone()`, and that's why I wanted to use `hasTemporalInformation`. unfortunately, I haven't been able to think of a cleaner way to do this

Reply:
OK, let's leave it this way for now. If you have a sec, throw in a comment about the Java version